File: src/easy/find_largest_altitude.zig
const std = @import("std");

const Allocator = std.mem.Allocator;

pub fn findLargestAltitude(comptime T: type, allocator: Allocator, gain: []const T) ?T {
    var result = allocator.alloc(T, gain.len + 1) catch return null;
    @memset(result, 0);
    var altitude: T = 0;
    for (gain, 0..) |height, i| {
        altitude += height;
        result[i] = altitude;
    }
    defer allocator.free(result);
    return std.mem.max(T, result);
}

File: src/main/java/easy/FindLargestAltitude.java
package easy;

import java.util.Arrays;

public class FindLargestAltitude {
  public int solve(int[] gain) {
    int altitude = 0;
    int[] result = new int[gain.length + 1];
    for (int i = 0; i < gain.length; i++) {
      result[i] = gain[i] + altitude;
      altitude += gain[i];
    }
    return Arrays.stream(result).max().getAsInt();
  }
}
